dotnet/src/SemanticKernel/Orchestration/Plan.cs

```diff
@@ -22,6 +22,12 @@ namespace Microsoft.SemanticKernel.Orchestration;
 /// </summary>
 public sealed class Plan : ISKFunction
 {
+    /// <summary>
+    /// The Schema version of the plan.
+    /// </summary>
+    [JsonPropertyName("schema")]
+    public int Schema { get; set; } = CurrentSchemaVersion;
```

Comment:
what's the design about schemas? how are we detecting breaking changes, documenting them, handling upgrades, etc? do we have tests to ensure schemas are not broken?

Reply:
There's more details to figure out, this is a stake in the ground similar to PromptTemplateConfig to make sure we can easily identify the serialization format of the plan and handle any changes in the future.

The schema version is used to validate the JSON serialization and deserialization of the plan, and an exception is thrown if the schema version does not match the current supported version. 

At the time of authoring, it wasn't a breaking change since this is all new. That's changed, but I think the impact is negligible right now, still. In the future, these things will be important, agreed.

The PR adds unit tests for both the Plan and ContextVariables classes, which cover various scenarios and edge cases of serializing and deserializing plans and context variables, including checks for invalid schema versions and invalid JSON strings.

Reply:
Will comment out schema work for now and leave for future/finalization to leave this PR to focus on improvements.

Reply:
Remove schema changes for now.